File: src/main/java/io/github/biligoldenwater/midiplayer/MidiPlayer.java
package io.github.biligoldenwater.midiplayer;

import io.github.biligoldenwater.midiplayer.utils.MidiPlay;
import org.bukkit.plugin.java.JavaPlugin;

import java.io.File;

public final class MidiPlayer extends JavaPlugin {
    private static MidiPlayer instance;

    private boolean debug = false;

    @Override
    public void onLoad() {
        instance = this;
    }

    @Override
    public void onEnable() {
        // Plugin startup logic
        getLogger().info("Enabled.");//输出已启用消息到日志
    }

    public static void main(String[] args) {
//        File file = new File("D:\\Music\\Midis\\U.N.オーエンは彼女なのか？.mid");
//        File file = new File("D:\\Music\\Midis\\When Christmas comes to town.mid");
//        File file = new File("D:\\Music\\Midis\\千本樱.mid");
        File file = new File("D:\\Music\\Midis\\HedwigsTheme.mid");
        new MidiPlay(file).run();
    }

    @Override
    public void onDisable() {
        // Plugin shutdown logic
        getLogger().info("Disabled.");//输出已禁用消息到日志
    }

    public boolean isDebug() {
        return debug;
    }

    public void setDebug(boolean debug) {
        this.debug = debug;
    }

    public static MidiPlayer getInstance() {
        return instance;
    }
}

File: src/main/java/io/github/biligoldenwater/midiplayer/MidiPlayer.kt
package io.github.biligoldenwater.midiplayer

import io.github.biligoldenwater.midiplayer.listener.OnAsyncPlayerChatEvent
import io.github.biligoldenwater.midiplayer.utils.MidiPlay
import org.bukkit.Bukkit
import org.bukkit.plugin.java.JavaPlugin

lateinit var instance: MidiPlayer

class MidiPlayer : JavaPlugin() {
    val playList: MutableMap<String, MidiPlay> = mutableMapOf()
    var isDebug = true
    override fun onLoad() {
        instance = this
    }

    override fun onEnable() {
        // Plugin startup logic
        Bukkit.getPluginManager().registerEvents(OnAsyncPlayerChatEvent(), this)
        logger.info("Enabled.") //输出已启用消息到日志
    }

    override fun onDisable() {
        // Plugin shutdown logic
        playList.forEach { (key: String?, value: MidiPlay) -> value.stop() }
        logger.info("Disabled.") //输出已禁用消息到日志
    }
}